beams/www/job_application_upload/upload_doc.py

```diff
@@ -0,0 +1,152 @@
+            doc.is_form_submitted = bool(data.get("is_form_submitted"))
+            doc.education_qualification = []
+            doc.professional_certification = []
+            doc.prev_emp_his = []
+            doc.language_proficiency = []
+            for row in data.get("educational_qualification", []):
+                if row.get('name_of_course_university'):
+                    filename = update_file(row.get('attachments'), 'Job Applicant', docname) or ''
+                    doc.append("education_qualification", {
+                        "name_of_course_university": row.get('name_of_course_university'),
+                        "name_location_of_institution": row.get('name_location_of_institution'),
+                        "dates_attended_from": int(row.get('dates_attended_from')),
+                        "dates_attended_to": int(row.get('dates_attended_to')),
+                        "result": float(row.get('result')),
+                        "attachments": filename
+                    })
+            for row in data.get("professional_certification", []):
+                if row.get("course"):
+                    filename = update_file(row.get('attachments'), 'Job Applicant', docname) or ''
+                    doc.append("professional_certification", {
+                        "course": row.get("course"),
+                        "institute_name": row.get("institute_name"),
+                        "dates_attended_from": int(row.get("dates_attended_from")),
+                        "dates_attended_to": int(row.get("dates_attended_to")),
+                        "type_of_certification": row.get("type_of_certification"),
+                        "subject_major": row.get("subject_major"),
+                        "attachments": filename
+                    })
+            for row in data.get("prev_emp_his", []):
+                if row.get("name_of_org"):
+                    filename = update_file(row.get('attachments'), 'Job Applicant', docname) or ''
+                    doc.append("prev_emp_his", {
+                        "name_of_org": row.get("name_of_org"),
+                        "prev_designation": row.get("prev_designation"),
+                        "last_salary_drawn": float(row.get("last_salary_drawn")),
+                        "name_of_manager": row.get("name_of_manager"),
+                        "period_of_employment": float(row.get("period_of_employment")),
+                        "reason_for_leaving": row.get("reason_for_leaving"),
+                        "attachments": filename
+                    })
+            for row in data.get("language_proficiency", []):
+                doc.append("language_proficiency", {
+                    "language": row["language"],
+                    "speak": row["speak"],
+                    "read": row["read"],
+                    "write": row["write"]
+                })
+            doc.flags.ignore_validate = True
+            doc.flags.ignore_mandatory = True
```

Comment:
is this necessary? if yes make sure ignoring mandatory and validations won't affect the flow